### ISSUE TITLE
[feature] Add "Connect under Reset" option

### ISSUE THREAD

**Issue by Gamadril** [stlink-org/stlink#577]
- [X] Programmer/board type: e.g Stlink/v2-onboard
- [x] Programmer firmware version: e.g STSW-LINK007 2.27.15
- [X] Operating system: Mac OS X
- [X] Stlink tools version and/or git commit hash: 1.3.1-9-gc04df7f
- [X] Stlink commandline tool name: `st-flash`
- [X] Target chip (and optional board): e.g STM32F030F4

SWCLK and SWDIO pins of my mcu are used and configured as output. That means to program the flash I have to use the "Connect Under Reset" option. This option seems to be missing. At least adding --reset does not help in that case.

**Comment by xor-gate** [comment]
Connect under reset is not to difficult the stlink has a command to assert the nRST pin and then try to connect. But this needs to be implemented.

**Comment by iglooom** [comment]
I have a same trouble and use this workaround [https://github.com/iglooom/stlink/commit/19381ff2e16a121a98184f58dc88838bcb25efd1](url)
It is not fully "connect under reset", but working for me.

**Comment by xor-gate** [comment]
The nRST reset pin is assert differently. I'm not sure what the output state of the pin is in your commit. First we need to measure and then implement a fix which works for everybody.

**Comment by geoffreymbrown** [comment]
I also needed this functionality.   While I added it to my own copy of the repo, it'd be nice if it made it in somehow.   Here are the issues in stlink_open_usb:

1)  Reset has to be asserted before any mode change can happen
2)  It's somewhat less than optimal that stlink_load_device_params(sl); happens in this routine, since loading flash parameters doesn't seem to work under reset.

Here's what I changed --
1)  reset parameter from bool to number :   values 0,1 keep current behavior.   reset == 2 is used to change the following code : 

` if (stlink_current_mode(sl) != STLINK_DEV_DEBUG_MODE) stlink_enter_swd_mode(sl);

    if (reset) {
        if ( sl->version.stlink_v > 1) stlink_jtag_reset(sl, 2);
        stlink_reset(sl);
        usleep(10000);
    }`

To:
 
`
 if (reset == 2)  stlink_jtag_reset(sl,0);
 if (stlink_current_mode(sl) != STLINK_DEV_DEBUG_MODE) stlink_enter_swd_mode(sl);

    if (reset == 1) {
        if ( sl->version.stlink_v > 1) stlink_jtag_reset(sl, 2);
        stlink_reset(sl);
        usleep(10000);
    }`
